core/cache/controller.go

```diff
@@ -198,10 +202,19 @@ func (c *Controller) Sweep() {
 	case <-c.manager.sweep():
 	case <-c.tomb.Dying():
 	}
+
+	select {
+	case <-c.initializing:
+		// The channel is already closed.
+	default:
+		close(c.initializing)
+	}
+
 	c.mu.Lock()
```

Comment:
I just realized quite an issue.

We call Sweep in the model cache for every change. We don't want to update the summaries every change.

If the initializing channel is already closed, we should return from the Sweep method.

Reply:
I'll sort that out.

Reply:
@howbazaar I've done this, but there is subtlety to consider.

I have not modified the cache starting state from what your initial logic was. This means that it does *not* come up in the initialising state.

The worker logic runs `Mark` immediately, so the in-theatre behaviour is as we desire - no summaries are generated until the cache is primed, *But* in the summary watcher tests do not use the worker, so they rely on an initialised starting state.

Technically, we should probably start the cache in the initialising state, but this would require adding an initial mark/sweep to about 14 of the tests. I'll leave this for your consideration.